cli/start.go

```diff
@@ -72,70 +76,98 @@ func MakeStartCommand() *cobra.Command {
 				// TODO-ACP: Infuture when we add support for the --no-acp flag when admin signatures are in,
 				// we can allow starting of db without acp. Currently that can only be done programmatically.
 				// https://github.com/sourcenetwork/defradb/issues/2271
-				opts = append(opts, node.WithACPPath(rootDir))
+				nodeOpts = append(nodeOpts, node.WithACPPath(rootDir))
 			}
 
 			if !cfg.GetBool("keyring.disabled") {
 				kr, err := openKeyring(cmd)
 				if err != nil {
 					return NewErrKeyringHelp(err)
 				}
+
 				// load the required peer key
 				peerKey, err := kr.Get(peerKeyName)
 				if err != nil {
 					return NewErrKeyringHelp(err)
 				}
-				opts = append(opts, net.WithPrivateKey(peerKey))
+				nodeOpts = append(nodeOpts, net.WithPrivateKey(peerKey))
+
 				// load the optional encryption key
 				encryptionKey, err := kr.Get(encryptionKeyName)
 				if err != nil && !errors.Is(err, keyring.ErrNotFound) {
 					return err
 				}
-
-				opts = append(opts, node.WithBadgerEncryptionKey(encryptionKey))
+				storeOpts = append(storeOpts, node.WithBadgerEncryptionKey(encryptionKey))
 
 				sourceHubKeyName := cfg.GetString("acp.sourceHub.KeyName")
 				if sourceHubKeyName != "" {
 					signer, err := keyring.NewTxSignerFromKeyringKey(kr, sourceHubKeyName)
 					if err != nil {
 						return err
 					}
-					opts = append(opts, node.WithTxnSigner(immutable.Some[node.TxSigner](signer)))
+					nodeOpts = append(nodeOpts, node.WithTxnSigner(immutable.Some[node.TxSigner](signer)))
 				}
 			}
 
 			acpType := cfg.GetString("acp.type")
 			if acpType != "" {
-				opts = append(opts, node.WithACPType(node.ACPType(acpType)))
+				nodeOpts = append(nodeOpts, node.WithACPType(node.ACPType(acpType)))
+			}
+
+			signalCh := make(chan os.Signal, 1)
+			signal.Notify(signalCh, os.Interrupt, syscall.SIGTERM)
+
+			for _, o := range storeOpts {
+				nodeOpts = append(nodeOpts, any(o))
 			}
 
-			n, err := node.NewNode(cmd.Context(), opts...)
+		START:
+			n, err := node.NewNode(cmd.Context(), nodeOpts...)
```

Comment:
todo: Please add a top level message, if defra is in development mode, warning the user to not use this configuration in production.

Reply:
done